### ISSUE TITLE
[member-naming] different patterns for constructor variables

### ISSUE THREAD

**Issue by Shinruchan** [typescript-eslint/typescript-eslint#1049]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/member-naming": ["error", { "private": "^_" }]
  }
}
```

```TS
export class Example {
  private _variable: string;

  constructor(private service: Service) {}
}
```

**Expected Result**

No errors.

**Actual Result**

Errors due to constructor generated variable does not match pattern.

**Additional Info**

Additional parameters for customizing constructor variables would be a great asset to this rule.

Having ability to customize constructor variables and class variables separately could prove to be useful and it would also simplify upgrade process from < 2.1.0 version to latest version.

Proposed additional settings that would inherit original settings if not set:
```JSON
"constructor": { "private": "", "protected": "", "public": "" }
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.3.3` |
| `@typescript-eslint/parser`        | `2.3.3` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `6.5.1` |
| `node`                             | `12.10.0` |
| `npm`                              | `6.11.3` |

**Comment by bradzacher** [comment]
Unless I'm mistaken - there's no options that would actually make this any easier, only options that would allow you to delay the code fix.
At some point we would remove those settings and you'd be in exactly the same place of having to correct the violating code.

As for adding those options as a permanent fixture - I don't think it's a good idea to add support for special casing the naming of parameter properties. Parameter properties are just properties defined in a different spot.
It was an oversight, and a bug that they were never handled.

**Comment by Shinruchan** [comment]
Problem did arrive from an upgrade perspective, and amount of the code needed to be corrected made me feel like this bugfix was more of a new feature. And then I wondered why isn't there any backwards compatibility for it.
It would for sure made it a lot easier to upgrade. But let's set this aside, because this is just a lazy man's view of the problem.

Even tho parameter properties are just properties as any other, they tend to be semantically different. They are usually outside dependencies, whether be it different services like in angular or just casual objects passed down via constructor. That fact would make sense to introduce different naming convention for them. 
It might increase readability of the code and help you spot potential errors if those variables were names in similar fashion.

I am not sure if this would be a big help, but I guess that any error caught before production due to help of static analysis is a worthwhile investment.
If you do not agree, feel free to close the issue.
